### ISSUE TITLE
3.2.1 minified version is broken

### ISSUE THREAD

**Issue by englishextra** [zengabor/zenscroll#22]
Please test before you publish.

https://raw.githubusercontent.com/zengabor/zenscroll/dist/zenscroll-min.js

I wasted 2 hours debugging

Your demo uses unminified version.

BTW Did you test 3.2.1 in Node environment? Browserify (my case) takes non min version and It produces an error with getBoundingClientRect. When remove browserify wrapping stuff, and symply adding a non min version -  the error still exists while i doesnt in regular browser.  Tested with "electronVersion": "1.3.2"

**Comment by zengabor** [comment]
What's the exact error you are getting?

I am not familiar with your environment (electron, browserify). Is there a `window` object and a `window.document` object in your environment?

**Comment by englishextra** [comment]
Here https://github.com/electron-userland/electron-prebuilt this applies to zenscroll.js

install electron - 

npm -g install electron-prebuilt

then create demo app https://github.com/atom-archive/electron-starter

and 

npm install zenscroll

get your zenscroll.js that comes after that action in appeared folder node_modules/zenscroll

then in index.html and embed zenscroll.js and in html write one of your inits

you may use

 var zenscroll = require("zenscroll");

zenscroll.to(0,500)

then you will get the error in debugger  - Ctrl+Shift+I

as for zenscroll-min.js it doesn't work in regular web developement environment - it appears you didnt read.

TO SUM UP

zenscroll.js works fine execpt at least in electron frame when the lib is required and then browserified - i didnt test when simply adding the lib code directly

zenscroll-min.js wont work wherever

**Comment by englishextra** [comment]
here

```
bundle.min.js:262 Uncaught TypeError: elem.getBoundingClientRect is not a function

return elem.getBoundingClientRect().top + getS

```

OK I see now whats the problem:

You shouldnt bother with testing on electron its because

this zenscroll.to(0,500); either this zenscroll.to(500,0); produces en error
these both are initiated from the start not wrapped in some condition or function

and this zenscroll.toY(500); doesn't

this happens with zenscroll 321 and is ok with zenscroll 100

In production https://english43.herokuapp.com/ I updated from 1.0.0 to 3.2.1

there is a piece of code - get it with webdeveloper plugin invormation - view javascript

w.zenscroll ? zenscroll.toY(hide_pos, 500) : w.scroll(0, hide_pos);

and it works as well as zenscroll.to(  https://englishextra.github.io/pages/articles/articles_reading_rules_utf.html

So it appears that .getBoundingClientRect is a problem.

I already faced that that in some context it wont work

.getBoundingClientRect has to be call when the elemnt exists and when .getBoundingClientRect is wrapped in a function it may throw an error.

Compare the usage/appearance of .getBoundingClientRect in zenscroll 100 and 321
https://gist.github.com/englishextra/a547ad3055144ca927a6ce32fa769d36
the trouble is here

```
return b?a.offsetTop:a.getBoundingClientRect().top+m()-e.offsetTop
```

The problem's there

Cheers

TO devs that enjoy and rely on zenscroll - I strongly recommend usage of zenscroll 1.0.0 its stable whereas 321 is unstable
We are in production - we cannot afford to be testing sites

**Comment by zengabor** [comment]
If you want to scroll to a specific position, then use `zenscroll.toY()`

`zenscroll.to()` will scroll to a specific element, e.g.,

```
var about = document.getElementById("about")
zenscroll.to(about)
```

You can see these described here: https://zengabor.github.io/zenscroll/#howtouse

**Comment by englishextra** [comment]
@zengabor Are you listening? Fix this https://github.com/zengabor/zenscroll/blob/dist/zenscroll-min.js - it produces an error. Just after inclusion.

**Comment by zengabor** [comment]
Can you give me an URL with the error so I can see it happening? Thanks

**Comment by englishextra** [comment]
jsfiddle.net/englishextra/gemL2ucp  Cheers

**Comment by zengabor** [comment]
Change `zenscroll.to(0,500)` to `zenscroll.toY(0,500)`

**Comment by ezamelczyk** [comment]
Sorry to say but your minified version is really broken.
Here is example with minified version:
https://jsfiddle.net/soberowy/78aj8gns/

And here with normal:
https://jsfiddle.net/soberowy/c1410t4w/

**Comment by zengabor** [comment]
Thank you very much, @soberowy 

My build process was broken and not the latest version was included in the commit. Sorry about that. Fixed in version 3.2.2
